### ISSUE TITLE
Using `der` on parameters

### ISSUE THREAD

**Issue by phannebohm** [OpenModelica/OpenModelica#8890]
### Description

When using the derivative operator on a parameter, the model fails in the runtime.

### MWE

```modelica
model ParameterDer
  parameter Real p;
  Real x;
equation
  der(p) = x;
end ParameterDer;
```

### Expected Behavior

The model should do one of two things:
* set `x = 0` and run without error, perhaps emit a warning.
* fail with a good error message, probably in the backend.

**Comment by casella** [comment]
Although I can't find any useful use case for such a construct, from a mathematical standpoint the derivative of a parameter is zero by definition. I can't see why this should be an error. 

A warning could be justified by the fact that der(p) plays no useful role, so it's probably a mistake, and it is good reporting that. But I'm not sure about this, I guess we could skip that.

Hence, I would suggest that the backend catches all derivatives of parameter expressions, and simply replace them by zero.

**Comment by perost** [comment]
The [specification](https://specification.modelica.org/master/operators-and-expressions.html#modelica:der) says:
> For Real parameters and constants the result is a zero scalar or array of the same size as the variable.

**Comment by phannebohm** [comment]
Should the frontend already do that substitution?

**Comment by perost** [comment]
> Should the frontend already do that substitution?

It could easily do that if we want, during simplification perhaps?

**Comment by perost** [comment]
Something like #8892 perhaps?

**Comment by perost** [comment]
Fixed with #8892.

**Comment by casella** [comment]
> The [specification](https://specification.modelica.org/master/operators-and-expressions.html#modelica:der) says:
> 
> > For Real parameters and constants the result is a zero scalar or array of the same size as the variable.

Shame on me, I should have checked 😄
